fairseq/models/transformer/transformer_encoder.py

```diff
@@ -202,9 +202,12 @@ def forward_scriptable(
         """
         # compute padding mask
         encoder_padding_mask = src_tokens.eq(self.padding_idx)
-        has_pads: Tensor = (
+        has_pads = (
             torch.tensor(src_tokens.device.type == "xla") or encoder_padding_mask.any()
         )
+        # Torchscript doesn't handle bool Tensor correctly, so we need to work around.
+        if torch.jit.is_scripting():
```

Comment:
Can you also add torch.jit.is_tracing()?

Reply:
@tugsbayasgalan after second thought I don't think we need to handle jit.tracing case. If you take a look at this line in the same function:
```
if return_all_hiddens and not torch.jit.is_scripting():
```

it doesn't account for tracing case as well. proactively adding a tracing guard here might not have any real use case.